Common/include/basic_types/ad_structure.hpp

```diff
@@ -676,6 +719,42 @@ FORCEINLINE void ResumePreaccumulation(bool wasActive) {
   SU2_OMP_SAFE_GLOBAL_ACCESS(PreaccEnabled = true;)
 }
 
+struct ErrorReport {
+  unsigned long ErrorCounter;
+  std::ofstream* out;
+};
+
+FORCEINLINE void ResetErrorCounter(struct ErrorReport* report) { report->ErrorCounter = 0; }
```

Comment:
Doesn't seem like there is a reason to pass these by pointer since the functions don't do any checking on whether the argument is valid.
```suggestion
FORCEINLINE void ResetErrorCounter(ErrorReport& report) { report.ErrorCounter = 0; }
```